### DISCUSSION TITLE
how to connect page with edit function

### DISCUSSION THREAD

**Discussion by gaborrubus** [neuefische/questions#597] [Web] [answered]
Hello,

I'm currently working on my edit function, and I am stuck. 
I am trying to use the find method, but it doesn’t find the ID of the object in the array.
If I click the edit button, it directs me to the detailed Page with a form, but it says that the ID is undefined.
I have added a useEffect too and I cannot  get the id from my router path and pass it to my handleEditActivity function in _app.js?

Example:
localhost:3000/0/edit (pages/[id]/edit.js)
-> I want to access the “0” and pass it to _app.js

I have committed and pushed my current status. This can be reached at this [link](https://github.com/gaborrubus/capstone-project/pull/15). The branch is called feature/userstory6/edit.

**Comment by mbosselmann** [comment, marked as answer]
Hi Gabor,
I checked your code and have the following solution for your edit functionality. You do not need a useEffect, you can use the id directly:


```jsx
export default function EditActionItemPage({ handleEditActivity }) {
  const router = useRouter();
  const { id } = router.query;

  return (
    <>
      <Header>Edit Adventure</Header>
      <StyleList />
      <FormActivityForEdit
        onSubmit={(editedActivity) => handleEditActivity(editedActivity, id)}
      />
    </>
  );
}
```

But you need to adapt your `handleEditActivity` function so that it can handle two arguments: `editedActivity` and `id`. :)
In your `_app.s` should your function look like this then:

```jsx
  function handleEditActivity(editedActivity, id) {
    const { name, points, description } = editedActivity;

    const foundActivity = activity.find((activity) => activity.id === id);
    if (!foundActivity) {
      console.log(`Activity with id ${id} not found`);
      return;
    }
    const updatedActivity = {
      ...foundActivity,
      name,
      points,
      description,
    };
    const updatedActivities = activity.map((activity) =>
      activity.id === id ? updatedActivity : activity
    );

    setActivity(updatedActivities);
    setUpdatedActivity(updatedActivity);
    router.push(`/${id}`);
    setActivityId(id);
  }
```

It is very important too to set the correct path for `router.push`. `activity/${id}`is a route that you did not define.

I hope this does help. :)

PS: It would be really cool to see the values of the object the user wants to edit in the form. But I think that is the next step of your feature? I am looking forward to it. 👀

**Reply by gaborrubus** [reply]
THANK YOUUUUUUU!!!! It helped! And yes, the values for editing will be in the form:)
